.github/workflows/dvc-diff.yml

```diff
@@ -0,0 +1,53 @@
+# Checks for image diffs in a Pull Request and adds a GitHub comment showing the diff
+name: DVC image diff
+
+on:
+  pull_request:
+    paths:
+      - 'pygmt/tests/baseline/*.png.dvc'
+
+jobs:
+  dvc-diff:
+    name: DVC image diff
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+      with:
+        # fetch all history so that git diff works
+        fetch-depth: 0
+
+    - name: Setup data version control (DVC)
+      uses: iterative/setup-dvc@v1
+
+    - name: Setup continuous machine learning (CML)
+      uses: iterative/setup-cml@v1
+
+    # Produce the markdown diff report, which should look like:
+    # | Status   | Path                                |
+    # |----------|-------------------------------------|
+    # | added    | pygmt/tests/baseline/test_image.png |
+    - name: Get list of images that were added or changed
+      run: |
+        dvc diff --show-md master HEAD >> report.md
+        cat report.md
+
+    - name: Pull image data from cloud storage
+      run: dvc pull --remote upstream
```

Comment:
Currently this line can only pull from the upstream dvc remote at https://dagshub.com/GenericMappingTools/pygmt. Will need to think about how to make it work for forks as well.

Reply:
Do we need to care about forks, if people follow the contributing guides (i.e., asking for permission on DAGsHub)? If people fork the DAGsHub repository and upload DVC images to their fork, our Tests workflow will fail, because it can't download the DVC images, right?

Reply:
I think we can deal with fork on the second iteration of this dvc-diff-action workflow. Technically we should be able to do `dvc pull` from DAGsHub forks, but that'll be quite a bit of work to code up as we need to point to another DVC remote (and also not sure if this is the best way to do things).

Reply:
I believe very few users will work with DAGsHub forks, so not a big issue for us.

Reply:
Ok, will kick the can down the road then.